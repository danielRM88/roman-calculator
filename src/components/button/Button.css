.button-style {
  border: 1px solid black;
  display: block;
  width: 100%;
  height: 60px;
  font-size: 1em;
  color: white;
  background-color: #49505f;
}
