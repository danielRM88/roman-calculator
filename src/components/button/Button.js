import React from "react";
import "./Button.css";

function Button({ label, onActionClick }) {
  return (
    <button
      data-testid={"button-" + label}
      className="button-style"
      onClick={() => onActionClick(label)}
    >
      {label}
    </button>
  );
}

export default Button;
