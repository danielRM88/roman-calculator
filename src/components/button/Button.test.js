import React from "react";
import ReactDOM from "react-dom";
import Button from "./Button";

import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

describe("Button component", () => {
  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div);
  });

  it("Renders button correctly", () => {
    const { getByTestId } = render(<Button label="test"></Button>);
    const label = getByTestId("button");
    expect(label).toHaveTextContent("test");
  });

  // Snapshot testing
  it("matches snapshot", () => {
    const tree = renderer.create(<Button label="save"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
