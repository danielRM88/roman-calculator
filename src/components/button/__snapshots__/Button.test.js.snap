// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<button
  className="button-style"
  data-testid="button"
>
  save
</button>
`;
